nagios/datadog_checks/nagios/nagios.py

```diff
@@ -218,12 +231,15 @@ def check(self):
             self.log.debug("Start nagios check for file %s", self.log_path)
             next(self.gen)
             self.log.debug("Done nagios check for file %s (parsed %s line(s))", self.log_path, self._line_parsed)
-        except StopIteration as e:
+        except (RuntimeError, StopIteration) as e:
             self.log.exception(e)
-            self.log.warning("Can't tail %s file", self.log_path)
+            self.log.warning(
+                "Can't tail %s file, will retry from the end of file on the next check run.", self.log_path
+            )
+            self._reset()
```

Comment:
Previously when there is a failure the `agent status` looks like this:

```

    nagios (1.6.1)
    --------------
      Instance ID: nagios:3d63b64aaf0f3ee7 [ERROR]
      Configuration Source: file:/etc/datadog-agent/conf.d/nagios.d/nagios.yaml
      Total Runs: 3
      Metric Samples: Last Run: 0, Total: 26
      Events: Last Run: 0, Total: 6
      Service Checks: Last Run: 0, Total: 0
      Average Execution Time : 19ms
      Last Execution Date : 2020-09-01 14:23:38.000000 UTC
      Last Successful Execution Date : 2020-09-01 14:23:23.000000 UTC
      Error: generator raised StopIteration
      Traceback (most recent call last):
        File "/opt/datadog-agent/embedded/lib/python3.8/site-packages/datadog_checks/base/utils/tailfile.py", line 113, in tail
          self._open_file(move_end=False, pos=pos)
        File "/opt/datadog-agent/embedded/lib/python3.8/site-packages/datadog_checks/base/utils/tailfile.py", line 33, in _open_file
          stat = os.stat(self._path)
      FileNotFoundError: [Errno 2] No such file or directory: '/opt/nagios/var/log/nagios.log'

      During handling of the above exception, another exception occurred:

      Traceback (most recent call last):
        File "/opt/datadog-agent/embedded/lib/python3.8/site-packages/datadog_checks/base/utils/tailfile.py", line 118, in tail
          raise StopIteration(e)
      StopIteration: [Errno 2] No such file or directory: '/opt/nagios/var/log/nagios.log'

      The above exception was the direct cause of the following exception:

      Traceback (most recent call last):
        File "/opt/datadog-agent/embedded/lib/python3.8/site-packages/datadog_checks/base/checks/base.py", line 827, in run
          self.check(instance)
        File "/opt/datadog-agent/embedded/lib/python3.8/site-packages/datadog_checks/nagios/nagios.py", line 192, in check
          tailer.check()
        File "/opt/datadog-agent/embedded/lib/python3.8/site-packages/datadog_checks/nagios/nagios.py", line 219, in check
          next(self.gen)
      RuntimeError: generator raised StopIteration
```

But with this PR, no error is seen on `agent status`. 

I think we should keep report the error in `agent status` by propagating up the exception for `RuntimeError` type at least, not sure if we should also do so for `StopIteration`.

Reply:
We probably don't want to fail the check altogether if one log file is missing. I can use `self.warning` though, WDYT?

Reply:
That's a good point. Agreed. Let's use `self.warning` then.